### ISSUE TITLE
allow ignoring tests

### ISSUE THREAD

**Issue by amtoine** [nushell/nupm#46]
i have a test in `nu-git-manager` that cannot run for now and thus i thought it would be cool to ignore it.

## Description
for now, the only way to ignore a test is to not `export` it.
however, by doing so, the test won't appear in the output of `nupm test`...

i think it would be cool to support something like
```nushell
# tests/mod.nu
export def my-test-ignore [] {
    # ...
}
```
and still have `my-test` appear in yellow next to the other ones :yum: 

```
Testing package /path/to/my-package
tests test-fail-1 ... FAIL
tests test-fail-3 ... FAIL
tests test-success ... SUCCESS
tests test-ignore-1 ... IGNORE
tests test-fail-2 ... FAIL
tests test-ignore-2 ... IGNORE
Ran 5 tests. 1 succeeded, 2 ignored, 3 failed.
```

**Comment by NonlinearFruit** [comment]
A big draw back to using the test name is if you have a test like `ignore-node-modules` or similar would be skipped

**Comment by vyadh** [comment]
I didn't actually see what nupm was doing to run tests (I initially assumed it was written in Rust) but I see it's doing similar to [my recent PR](https://github.com/nushell/nushell.github.io/pull/1594/files) to the documentation that illustrates a tiny embedded test runner method that does something very similar.

That ignores tests like this:

```nu
# ignore
def test_fib_ignored_test [] {
  print "This test will not be executed"
}
```

by doing

```nu
 scope commands
      | where ($it.name | str starts-with "test_")
      | where not ($it.description | str starts-with "ignore")
     # ...
```

I imagine there is a whole bunch of stuff that could be driven from description instructions. E.g. running a test method with different test data.

**Comment by Mrfiregem** [comment]
With attributes I think this would be pretty ergonomic. Here's a [proof-of-concept implementation](https://github.com/Mrfiregem/nupm/commit/0124b4b92048f3a72bf0a73ba15547df4c3b5d1c) by filtering on tests annotated with `@category 'ignore'`.

**Comment by 132ikl** [comment]
it would probably be better to use an attribute specifically for this use case instead of repurposing category. Ex:
```nushell
export alias "attr ignore" = echo

@ignore
export def ignore-test [] {
    assert equal true false
}
```

**Comment by Mrfiregem** [comment]
Yeah, I mentioned that in the commit message. There's the extra overhead of defining the command in the tests module, but maybe it's better.
